pkg/secretsmanager/secretsmanager.go

```diff
@@ -129,43 +105,180 @@ func NewSecretManager(ctx context.Context, instance *v1alpha1.SecretAgentConfigu
+	if config.CredentialsSecretName != "" {
+		// load credentials secret from Kubernetes secret
+		secObject, err := LoadCredentialsSecret(rClient, config, cloudCredNS)
+		if err != nil {
+			return &secretManagerGCP{}, err
+		}
+
+		// Create temporary dir for gcp credentials if needed
+		dir, err := ioutil.TempDir("", "cloud_credentials-*")
+		if err != nil {
+			log.Error(err, "couldn't create a temporary credentials dir")
+			return &secretManagerGCP{}, err
+		}
+
+		// clean up after ourselves
+		defer os.RemoveAll(dir)
+
+		// open a new file for writing
+		writeFile := func(name string, contents []byte) (string, error) {
+			fPath := path.Join(dir, name)
+
+			// open a new file for writing only
+			file, err := os.OpenFile(
+				fPath,
+				os.O_WRONLY|os.O_TRUNC|os.O_CREATE,
+				0666,
+			)
+			if err != nil {
+				return "", err
+			}
+
+			// close file when finished
+			defer file.Close()
+
+			// write bytes to file
+			_, err = file.Write(contents)
+			if err != nil {
+				return "", err
+			}
+			return fPath, nil
+		}
+
+		// get credential data
+		keyValue, err := getCredentialData(secObject)
+		if err != nil {
+			return &secretManagerGCP{}, err
+		}
+
+		// write credential data to file
+		fp, err := writeFile("gcp_credentials.json", keyValue)
+		if err != nil {
+			log.Error(err, "couldn't write credential data to file")
+			return &secretManagerGCP{}, err
+		}
+
+		// Create client with credentials file
+		client, clientErr = secretmanager.NewClient(ctx, option.WithCredentialsFile(fp))
+	} else {
+		// Create client without credentials file
+		client, clientErr = secretmanager.NewClient(ctx)
+	}
+
+	if clientErr != nil {
+		log.Error(clientErr, "couldn't create Google Secret Manager client")
+		return &secretManagerGCP{}, clientErr
 	}
 
 	return &secretManagerGCP{
 		client:               client,
-		secretsManagerPrefix: conf.SecretsManagerPrefix,
-		projectID:            conf.GCPProjectID,
-	}, err
+		secretsManagerPrefix: config.SecretsManagerPrefix,
+		projectID:            config.GCPProjectID,
+	}, nil
 }
 
 // newAWS configures a AWS secret manager object
-func newAWS(ctx context.Context, conf *v1alpha1.AppConfig) (context.Context, *secretManagerAWS) {
-	client := awssecretsmanager.New(session.New(&aws.Config{Region: aws.String(conf.AWSRegion)}))
+func newAWS(ctx context.Context, config *v1alpha1.AppConfig, rClient client.Client, cloudCredNS string) (context.Context, *secretManagerAWS, error) {
+	var accessKey string
+	var secretAccessKey string
+	var client *awssecretsmanager.SecretsManager
+
+	// if credentials secret is provided
+	if config.CredentialsSecretName != "" {
+		// load credentials secret from Kubernetes secret
+		secObject, err := LoadCredentialsSecret(rClient, config, cloudCredNS)
+		if err != nil {
+			return nil, &secretManagerAWS{}, err
+		}
+
+		// extract access AWS_ACCESS_KEY_ID from Kubernetes secret
+		if keyValue, ok := secObject.Data[string(v1alpha1.SecretsManagerAwsAccessKeyID)]; ok {
```

Comment:
Is "not ok" handled?

Reply:
ah yes, I missed that. I'll submit a commit handling those.